cmd/executor/cmd/root.go

```diff
@@ -110,21 +113,34 @@ func checkDockerfilePath() error {
 	return errors.New("please provide a valid path to a Dockerfile within the build context")
 }
 
-// resolveSourceContext unpacks the source context if it is a tar in a GCS bucket
+// resolveSourceContext unpacks the source context if it is a tar in a bucket
 // it resets srcContext to be the path to the unpacked build context within the image
 func resolveSourceContext() error {
 	if srcContext == "" && bucket == "" {
-		return errors.New("please specify a path to the build context with the --context flag or a GCS bucket with the --bucket flag")
+		return errors.New("please specify a path to the build context with the --context flag or a bucket with the --bucket flag")
 	}
-	if bucket == "" {
-		return nil
-	}
-	logrus.Infof("Using GCS bucket %s as source context", bucket)
+
 	buildContextPath := constants.BuildContextDir
-	if err := util.UnpackTarFromGCSBucket(bucket, buildContextPath); err != nil {
+
+	// if no prefix use Google Cloud Storage as default for backwards compability
+	if !strings.Contains(srcContext, "://") {
+		if bucket != "" {
+			srcContext = fmt.Sprintf("gs://%s", bucket)
+		} else {
+			srcContext = fmt.Sprintf("dir://%s", srcContext)
+		}
+	}
+
+	contextExecutor, err := buildcontext.GetBuildContext(srcContext)
+	if err != nil {
+		return err
+	}
+
+	if err := contextExecutor.UnpackTarFromBuildContext(srcContext, buildContextPath); err != nil {
```

Comment:
We need to remove the prefix before passing in `srcContext`. We should probably do that in `GetBuildContext` and store the value in each of the build context structs